src/Security/BeamNG.cpp

```diff
@@ -158,6 +170,110 @@ void FileList(std::vector<std::string>& a, const std::string& Path) {
         }
     }
 }
+
+#if defined(__APPLE__)
+
+std::string GetBottlePath() {
+    return BottlePath;
+}
+
+std::string GetBottleName() {
+    std::filesystem::path bottlePath(BottlePath);
+    return bottlePath.filename().string();
+}
+
+std::map<std::string, std::string> GetDriveMappings(const std::string& bottlePath) {
+    std::map<std::string, std::string> driveMappings;
+    std::string dosDevicesPath = bottlePath + "/dosdevices/";
+
+    if (std::filesystem::exists(dosDevicesPath)) {
+        for (const auto& entry : std::filesystem::directory_iterator(dosDevicesPath)) {
+            if (entry.is_symlink()) {
+                std::string driveName = Utils::ToLower(entry.path().filename().string());
+                driveName.erase(std::remove(driveName.begin(), driveName.end(), ':'), driveName.end());
+                std::string macPath = std::filesystem::read_symlink(entry.path()).string();
+                if (!std::filesystem::path(macPath).is_absolute()) {
+                    macPath = dosDevicesPath + macPath;
+                }
+                driveMappings[driveName] = macPath;
+            }
+        }
+    } else {
+        error("Failed to find dosdevices directory for bottle '" + bottlePath + "'");
+    }
+    return driveMappings;
+}
+
+bool CheckForGame(const std::string& libraryPath, const std::map<std::string, std::string>& driveMappings) {
+    //Convert the Windows path to Unix path
+    char driveLetterChar = std::tolower(libraryPath[0]);
+    std::string driveLetter(1, driveLetterChar);
+
+    if (!driveMappings.contains(driveLetter)) {
+        warn("Drive letter " + driveLetter + " not found in mappings.");
+        return false;
+    }
+
+    std::filesystem::path basePath = driveMappings.at(driveLetter);
+    debug("Base path: " + basePath.string());
+
+
+    std::filesystem::path cleanLibraryPath = std::filesystem::path(libraryPath.substr(2)).make_preferred();
+    std::string cleanPathStr = cleanLibraryPath.string();
+    std::replace(cleanPathStr.begin(), cleanPathStr.end(), '\\', '/');
+    cleanLibraryPath = std::filesystem::path(cleanPathStr);
+    if (!cleanPathStr.empty() && cleanPathStr[0] == '/') {
+        cleanPathStr.erase(0, 1);
+    }

```

Comment:
Is this always the case? How can `libraryPath` look? If it always has the leading `/` wouldn't it be better to strip 3 characters instead of 2 where you do `libraryPath.substr(2)`?

Reply:
The goal is to get a path that not starting with `/` or ending with `/`.
Then I update the code.